client/xdag_config.c

```diff
@@ -0,0 +1,535 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+//#include <errno.h>
+#include "xdag_config.h"
+
+
+
+#ifndef true
+#define true (1)
+#endif
+
+#ifndef false
+#define false (0)
+#endif
+
+#if defined (WIN32)
+#define strkscmp stricmp
+#else
+#define strkscmp strcasecmp
+#endif
+
+static char *trim(char *str)
+{
+    if(str != NULL)
+    {
+        int len = (int)strlen(str);
+        char *start = str;
+        char *end = str + len;
+        while(start < end && *start <= ' ')
+        {
+            start++;
+        }
+        while(end > start && *(end-1) <= ' ')
+        {
+            end--;
+        }
+        if(end - start != len)
+        {
+            strncpy(str, start, end-start);
+            str[end-start] = '\0';
+        }
+    }
+    return str;
+}
+
+
+
+
+
+typedef struct
+{
+    char *name;
+    char *value;
+} KEY;
+
+typedef struct
+{
+    char *name;
+    int number_keys;
+    KEY **keys;
+    char **key_names;
+} NODE;
+
+
+
+typedef struct
+{
+    char *path;
+    int number_nodes;
+    NODE **nodes;
+    char **node_names;
+} CFG;
+
+
+static const char COMMENT_MARK[] = {'#', '*'};
+typedef enum
+{
+    TYPE_EMPTYLINE,
+	TYPE_COMMENT,
+    TYPE_NODE,
+    TYPE_KEY,
+    TYPE_UNKNOW,
+} CFG_CONTENT_TYPE;
+
+static char *readline(FILE *fp)
+{
+    char buf[128];
```

Comment:
can you write an error in logs if line is longer than buff ? 
xdag_err function

Reply:
> 如果行长于buff，你能在日志中写错误吗？
> xdag_err函数

The line exceeds 128 characters and will not report an error. The length of the line is dynamically calculated in an iterative manner.

Reply:
ah i see, great